modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/pagination/Page.java

```diff
@@ -155,11 +159,12 @@ private Page(
 
 	private Page(
 		Map<String, Map<String, String>> actions, Collection<T> items,
-		Pagination pagination, long totalCount, List<Facet> facets) {
+		Pagination pagination, long totalCount,
+		Map<String, List<Facet>> facetMap) {
```

Comment:
It could be a good candidate for a Decorator pattern (with actions we could have done the same) and let the Page class to be an abstract factory.

This way we separate different logic and avoid passing null when instantiating. It may help also with the alphabetical parameter order that may cause us problems.

Don't know if it worth it but just saying

Reply:
Well, I just noticed that this may complicate code generation so may be it is better to ignore it

Reply:
Yeah, that's why I avoided it because each new class forces us to create a new template :(

Reply:
I think a new template would not be necessary. I was talking about selecting the correct method of Page to call, because with the Decorator I was thinking in decorating the Page, not the resource